lib/plotutil.py

```diff
@@ -34,14 +37,25 @@ def plot(func, xstart, xend, step, color_name, xlabel, ylabel, theme, gui):
 		mplstyle.use('dark_background')
 	else:
 		mplstyle.use('default')
-
-	xvals = []
-	i = xstart
-	while i <= xend:
-		xvals.append(i)
-		i += step
-	yvals = create_y_values(func, xvals)
-
+	if plotflag == 1:
+		xvals = []
+		i = xstart
+		while i <= xend:
+			xvals.append(i)
+			i += step
+		yvals = create_y_values(func, xvals)
+	elif plotflag == 0:
+		print(func)
+		xvalss = func[1:func.find(']')]
+		yvalss = func[func.find(']')+2:len(func)-1]
+		xvals = map(float, xvalss.split(','))
+		yvals = map(float, yvalss.split(','))
+		if len(xvals) == len(yvals):
+			plt.plot(xvals, yvals, color=color_name, linewidth=2.0)
+			plt.xlabel(xlabel)
+			plt.ylabel(ylabel)
```

Comment:
@LoneOne123 you can do the plotting only once in the try catch block. This part is basically redundant. Moreover the color validity is not checked over here. Try executing ```python plotit.py -l [1,2,3,4][2,3,4,5] -c hklb```. The app will crash.